### ISSUE TITLE
Has anyone successfully migrated this app to a self hosted Parse Server?

### ISSUE THREAD

**Issue by pbmarcano** [ParsePlatform/AnyPhone#37]
^

**Comment by jackintosh7** [comment]
^

**Comment by damtsnkff** [comment]
@pbmarcano @jackintosh7 Yep, I did. Depending on your use, the cloud code is ready to use. (the swift as well). Do you face any issue ?

**Comment by pbmarcano** [comment]
I have ACL problems when trying to create new users... I believe related to https://github.com/ParsePlatform/parse-server/issues/1495... Knowing that you seemed to have done it successfully I am curious if you know something I don't about the cloud code.

**Comment by damtsnkff** [comment]
@pbmarcano No I'm facing issues as well. I'm investigating, and keep you up to date. Do you receive a `Request timeout` message from the server ? In my case, it only happens after few request made by the new created user.

**Comment by jackintosh7** [comment]
I was still unable to get this working, I don't recall the error I was having. I eventually rebuilt it myself. Used Sinch verification. And only used the cloud code to change the user's password if they already have an account. I can post the code if interested, it's a nice workaround.

**Comment by damtsnkff** [comment]
@pbmarcano Ok so now it's working. I followed this issue https://github.com/ParsePlatform/parse-server/issues/1855 . (Added the `"mongodb":"2.1.18"` to my package.json) . 

<img width="1216" alt="screen shot 2016-05-25 at 12 28 28 pm" src="https://cloud.githubusercontent.com/assets/8188697/15537012/5cfbfb1e-2274-11e6-98e1-cd5685a5f86a.png">

> _Here is the log before and after doing this._ 


Also, here is what I did so far to make it works:
- I deleted all the `_User` and  `_Installation` previously created through Cloud Code . 
- I made sure the `_Installation` was created when user log-in, from the client side.
- I deleted the `user.setACL({});` from Cloud Code because the user by default is created with ACL. (at least, for the testing).

What I'm gonna try next : 
- Set the `user.setACL` from another Cloud Code function, once the user is created. (query first, then set the ACL and Role)
- Set `_Role`to the new created user.


So, where is your issue ? I'll try to reproduce it. Since it's with ACL, I'll concerned as well I guess.

**Comment by jamesjackson69** [comment]
@jackintosh7 It would be awesome if you could post your code!

**Comment by pbmarcano** [comment]
@damtsnkff sorry for not getting back to you sooner.

I tried adding monogdb 2.1.18 to package.json, as well as removing `user.setACL({});`. I am hesitant to delete the `_User` and `_Installation` class since the problem is coming from the database I share with my parse.com api that is still in production (which people are currently using).

Specifically here are the odd things that keep happening when a user is created with parse server cloud code. 

- the new user gets two sessions, one created at "signup"
`{"action":"signup","authProvider":"password"}` and one created at "login" `{"action":"login","authProvider":"password"}`. 
Both session ACLs are set to "Public Read + Write" instead of the User "objectID".
- On the User class, the user is created with an ACLs of Public Read, and their User object ID. 
- Attempting to log in as an existing user (rather than a new user) throws an error 206 on the iOS client suggesting I cannot modify the existing user.

When running the same code off Parse.com API cloud code only one session is created, with the right ACL, and the User ACL is simply just the objectID without public read.

I am a bit ignorant of all the technical differences between running cloud code off of parse.com and off of my own parse server, so I am unsure if there are any packages that I should be adding to package.json that aren't twilio, mongodb 2.1.18, or the default packages.

**Comment by pbmarcano** [comment]
@jackintosh7, I may be interested in that workaround... I've put too many hours into getting confused 😅

**Comment by jackintosh7** [comment]
No problem. I'll post it later today or tomorrow :)

**Comment by kss2153** [comment]
Do I need to download the twilio helper library into my parse server for this to work? I copied over the cloud code, but I'm getting an invalid function error.

**Comment by Abhishek-Mohan** [comment]
I would like a possible solution @jackintosh7 if possible?
On a side note, is anyone getting a problem related to `com.parse.ParseRequest$ParseRequestException: bad json response` when using the token from cloud code logIn.

**Comment by Morreke** [comment]
Is this working for anyone now and willing to share a tutorial ?
